pymc/distributions/timeseries.py

```diff
@@ -893,39 +911,149 @@ class EulerMaruyama(distribution.Continuous):
+        if size is not None:
+            batch_size = size
+        else:
+            batch_size = at.broadcast_shape(*sde_pars, init_dist)
+        init_dist = change_dist_size(init_dist, batch_size)
+
+        # Create OpFromGraph representing random draws from SDE process
+        # Variables with underscore suffix are dummy inputs into the OpFromGraph
+        init_ = init_dist.type()
+        sde_pars_ = [x.type() for x in sde_pars]
+        steps_ = steps.type()
+
+        noise_rng = aesara.shared(np.random.default_rng())
+
+        def step(*prev_args):
+            prev_y, *prev_sde_pars, rng = prev_args
+            f, g = sde_fn(prev_y, *prev_sde_pars)
+            mu = prev_y + dt * f
+            sigma = at.sqrt(dt) * g
+            next_rng, next_y = Normal.dist(mu=mu, sigma=sigma, rng=rng).owner.outputs
+            return next_y, {rng: next_rng}
+
+        y_t, innov_updates_ = aesara.scan(
+            fn=step,
+            outputs_info=[init_],
+            non_sequences=sde_pars_ + [noise_rng],
+            n_steps=steps_,
+            strict=True,
+        )
+        (noise_next_rng,) = tuple(innov_updates_.values())
+
+        sde_out_ = at.concatenate([init_[None, ...], y_t], axis=0).dimshuffle(
+            tuple(range(1, y_t.ndim)) + (0,)
+        )
+
+        eulermaruyama_op = EulerMaruyamaRV(
+            inputs=[init_, steps_] + sde_pars_,
+            outputs=[noise_next_rng, sde_out_],
+            dt=dt,
+            sde_fn=sde_fn,
+            ndim_supp=1,
+        )
+
+        eulermaruyama = eulermaruyama_op(init_dist, steps, *sde_pars)
+        return eulermaruyama
+
+
+@_change_dist_size.register(EulerMaruyamaRV)
+def change_eulermaruyama_size(op, dist, new_size, expand=False):
+
+    if expand:
+        old_size = dist.shape[:-1]
+        new_size = tuple(new_size) + tuple(old_size)
+
+    init_dist, steps, *sde_pars, _ = dist.owner.inputs
+    return EulerMaruyama.rv_op(
+        init_dist,
+        steps,
+        sde_pars,
+        dt=op.dt,
+        sde_fn=op.sde_fn,
+        size=new_size,
+    )
+
+
+@_logprob.register(EulerMaruyamaRV)
+def eulermaruyama_logp(op, values, init_dist, steps, *sde_pars_noise_arg, **kwargs):
+    (x,) = values
+    # noise arg is unused, but is needed to make the logp signature match the rv_op signature
+    *sde_pars, _ = sde_pars_noise_arg
+    # sde_fn is user provided and likely not broadcastable to additional time dimension,
+    # since the input x is now [..., t], we need to broadcast each input to [..., None]
+    # below as best effort attempt to make it work
+    sde_pars_broadcast = [x[..., None] for x in sde_pars]
+    xtm1 = x[..., :-1]
+    xt = x[..., 1:]
+    f, g = op.sde_fn(xtm1, *sde_pars_broadcast)
+    mu = xtm1 + op.dt * f
+    sigma = at.sqrt(op.dt) * g
+    # Compute and collapse logp across time dimension
+    sde_logp = at.sum(logp(Normal.dist(mu, sigma), xt), axis=-1)
+    init_logp = logp(init_dist, x[..., 0])
+    return init_logp + sde_logp
```

Comment:
Should we return a concatenation of `init_logp` and `sde_logp`

Reply:
> Should we return a concatenation of init_logp and sde_logp?

Nope because the time dimension is collapsed

> Also variable support dimensionality of init_logp might need to be handled, otherwise one might try to broadcast the other incorrectly.

I think it is fine as is, because:
1. init_logp is scalar and data is (n_batch, t), logp is broadcasted to (n_batch, )
2. init_logp is multivariate and data is (n_batch, t), logp is (n_batch, )

The only potential error is init_logp is multivariate, all the `sde_pars` are scalar and data is (t, ), maybe we can have a test to make sure the rank of `init_dist` is lower or equal to `sde_pars`, WDYT?

Reply:
If `init_logp` is multivariate, it must have support_shape `(n_batch[-1],)`. If `n_batch = (5,)` the logp will be a scalar, and you will be over counting it 5 times when you add it to the `sde_logp`.

Reply:
In the `AR` this is okay, because there are multiple initial points per timeseries (depending on the `ar_order`), so we can align them whether users pass scalar or multivariate `init_dist`. Does this distribution also have multiple `initial_points` per timeseries?

Reply:
> Does this distribution also have multiple initial points per timeseries?

It shouldn't. I am only thinking about `n_batch` being a scalar int here.
You are right (2) does not work because the logp will be `()`, maybe we should disable multivariate support for `init_dist` and add that later if needed.

Reply:
> You are right (2) does not work because the logp will be (), maybe we should disable multivariate support for init_dist and add that later if needed. WDYT?

Yes, I think you can't actually have a multivariate init_dist, otherwise the logp across multiple timeseries is no longer independent, and you would need to collapse those as well (after which you can indeed add the scalar with scalar)

Reply:
SG - I will update the doc string etc to make clear it should be (a batch of) scalar time series (and in the future we can make it into multivariate by having something like `sde_logp = at.sum(logp(MvNormal.dist(mu, cov), xt), axis=-1)`